atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/cleaner/KeyValueServicePuncherStore.java

```diff
@@ -36,16 +37,61 @@
 import com.palantir.atlasdb.table.description.ValueType;
 import com.palantir.atlasdb.transaction.api.ConflictHandler;
 import com.palantir.common.base.ClosableIterator;
+import com.palantir.exception.NotInitializedException;
+import com.palantir.processors.AutoDelegate;
 
 /**
  * A PuncherStore implemented as a table in the KeyValueService.
  *
  * @author jweel
  */
-public final class KeyValueServicePuncherStore implements PuncherStore {
+@AutoDelegate(typeToExtend = PuncherStore.class)
+public final class KeyValueServicePuncherStore implements PuncherStore, AsyncInitializer {
+    public static class InitializingWrapper implements AutoDelegate_PuncherStore {
```

Comment:
what if we moved all init code into this class, and made it non-static? then it seems like you could make `AsyncInitializer` an abstract class